### PR TITLE
chore: refactor user type as intersection of usernested

### PR DESCRIPTION
**Pull request by raastinn**

### CODE REVIEW

File: src/js/users/types.ts [hunk 1]
```diff
@@ -1,42 +1,68 @@
 import { AdministratorRoles } from "../administration/types";
 import { GroupMinimal, Permissions } from "../groups/types";
 
+/* Business to consumer specific details */
 type UserB2C = {
+    /* The display name (optional) */
     display_name?: string;
+    /* The family name (optional) */
     family_name?: string;
+    /* The given name (optional) */
```
Comment (ReeceHoffmann) [inline src/js/users/types.ts]:
```suggestion
    /* The given name  */
```
No need to add optional, should be obvious at a glance from the type itself. Same is true for other instances as well

File: src/js/users/types.ts [hunk 2]
```diff
@@ -1,42 +1,68 @@
...
 
-export type User = {
-    id: string;
+/* A user with the essential information */
+export type UserNested = {
+    /* Indicates if the user is an administrator */
     administrator: boolean;
+    /* The unique identifier */
+    id: string;
+    /* The user's handle or username */
```
Comment (ReeceHoffmann) [inline src/js/users/types.ts:22]:
```suggestion
    /* The display name */
```

File: src/js/users/types.ts [hunk 3]
```diff
@@ -1,42 +1,68 @@
...
     handle: string;
+};
+
+/* A sample user */
```
Comment (ReeceHoffmann) [inline src/js/users/types.ts]:
```suggestion
/* A virtool user */
```

File: src/js/users/types.ts [hunk 4]
```diff
@@ -1,42 +1,68 @@
...
+export type User = UserNested & {
+    /* Indicates if user is active */
```
Comment (ReeceHoffmann) [inline src/js/users/types.ts:28]:
```suggestion
    /* Whether the account is enabled */
```

File: src/js/users/types.ts [hunk 5]
```diff
@@ -1,42 +1,68 @@
...
     b2c?: UserB2C;
+    /* The user's display name (optional) */
     b2c_display_name?: string;
+    /* The user's family name (optional) */
     b2c_family_name?: string;
+    /* The user's given name (optional) */
     b2c_given_name?: string;
+    /* The user's object ID (optional) */
     b2c_oid?: string;
+    /* Indicates if the user is required to reset password */
```
Comment (ReeceHoffmann) [inline src/js/users/types.ts]:
```suggestion
    /* Whether the user will be forced to reset their password on next login*/
```

File: src/js/users/types.ts [hunk 6]
```diff
@@ -1,42 +1,68 @@
...
     force_reset: boolean;
+    /*  */
```
Comment (ReeceHoffmann) [inline src/js/users/types.ts]: ?

File: src/js/users/types.ts [hunk 7]
```diff
@@ -1,42 +1,68 @@
...
     groups: Array<GroupMinimal>;
+    /* The date of the user's last password change */
     last_password_change: Date;
+    /* The user's permissions */
     permissions: Permissions;
+    /* The user's primary group */
     primary_group: GroupMinimal;
+    /* THe role of the administrator user */
```
Comment (ReeceHoffmann) [inline src/js/users/types.ts]:
```suggestion
    /* Their administrator role defining what resources they can modify*/
```

File: src/js/users/types.ts [hunk 8]
```diff
@@ -1,42 +1,68 @@
...
     administrator_role: AdministratorRoles;
 };
 
+/* A response containing a list of users */
 export type UserResponse = {
+    /* The array of user objects */
```
Comment (ReeceHoffmann) [inline src/js/users/types.ts]:
```suggestion
    /* The page of users*/
```

File: src/js/users/types.ts [hunk 9]
```diff
@@ -1,42 +1,68 @@
...
     last_password_change: Date;
+    /* The user's permissions */
     permissions: Permissions;
+    /* The user's primary group */
     primary_group: GroupMinimal;
+    /* THe role of the administrator user */
     administrator_role: AdministratorRoles;
 };
 
+/* A response containing a list of users */
```
Comment (ReeceHoffmann) [inline src/js/users/types.ts]:
```suggestion
/* User search results from the API */
```

File: src/js/users/types.ts [hunk 10]
```diff
@@ -1,42 +1,68 @@
...
     b2c_family_name?: string;
+    /* The user's given name (optional) */
     b2c_given_name?: string;
+    /* The user's object ID (optional) */
     b2c_oid?: string;
+    /* Indicates if the user is required to reset password */
     force_reset: boolean;
+    /*  */
     groups: Array<GroupMinimal>;
+    /* The date of the user's last password change */
```
Comment (ReeceHoffmann) [inline src/js/users/types.ts]:
```suggestion
    /* The date of their last password change */
```

File: src/js/users/types.ts [hunk 11]
```diff
@@ -1,42 +1,68 @@
...
     active: boolean;
+    /* The B2C user information (optional) */
     b2c?: UserB2C;
+    /* The user's display name (optional) */
     b2c_display_name?: string;
+    /* The user's family name (optional) */
     b2c_family_name?: string;
+    /* The user's given name (optional) */
     b2c_given_name?: string;
+    /* The user's object ID (optional) */
```
Comment (ReeceHoffmann) [inline src/js/users/types.ts]:
```suggestion
    /* Their B2C object ID*/
```

File: src/js/users/types.ts [hunk 12]
```diff
@@ -1,42 +1,68 @@
...
+    id: string;
+    /* The user's handle or username */
     handle: string;
+};
+
+/* A sample user */
+export type User = UserNested & {
+    /* Indicates if user is active */
     active: boolean;
+    /* The B2C user information (optional) */
```
Comment (ReeceHoffmann) [inline src/js/users/types.ts]:
```suggestion
    /* Their B2C specific information*/
```

File: src/js/users/types.ts [hunk 13]
```diff
@@ -1,42 +1,68 @@
...
     b2c?: UserB2C;
+    /* The user's display name (optional) */
```
Comment (ReeceHoffmann) [inline src/js/users/types.ts]:
```suggestion
    /* Their display name (optional) */
```

... ect for other fields

**Review by ReeceHoffmann — CHANGES_REQUESTED** [review]

File: src/js/users/types.ts (hunk 2 above)
Comment (igboyes) [reply src/js/users/types.ts:22]: > 

It should be left as is. 'Display name' is not used anywhere else as a variable name or terminology.

File: src/js/users/types.ts (hunk 4 above)
Comment (igboyes) [reply src/js/users/types.ts:28]: Terminology is right here. Users are activated or inactived. Not enabled or disabled.

Comment (ReeceHoffmann) [reply src/js/users/types.ts:28]: `whether the user account can be logged into`? `active` alone could be interpreted as `currently online`

**Review by ReeceHoffmann — APPROVED** [review]

**Pull request merged**